### ISSUE TITLE
Camera's incorrect fade zone when static tile layer is in front

### ISSUE THREAD

**Issue by inmylo** [photonstorm/phaser#4917]
## Version

* Phaser Version:

<= 3.21
* Operating system:

Linux

## Description

If there exist 2 tilemap layers and static is **in front** of dynamic one (**setDepth**), run camera's fade effect - fade zone is static and does not move when camera moves. In the example below (a modified version of https://labs.phaser.io/view.html?src=src\game%20objects\tilemap\collision\tile%20callbacks.js) - I've added some camera shift to the right to see the bug better.

Use arrows to collect coins - fade effect will run and you can see that it's **position** is not where camera's is. If you continue to move to the right while collecting coins - fade zone won't move with camera and you'll see fade zone's right border in the middle of the map.

If you set both layers depth to **10** or static is below a dynamic one - no issue.

Happens with WEBGL only.

Expected result - fade zone follows camera and takes all visible tilemap.

## Example Test Code

```
var config = {
    type: Phaser.WEBGL,
    // type: Phaser.CANVAS, // <- This forks fine
    width:  800,
    height: 576,
    backgroundColor: '#555',
    physics: {
        default: 'arcade'
    },
    scene: {
        preload: preload,
        create:  create,
        update:  update
    }
};

var game = new Phaser.Game(config);
var map;
var cursors;
var debugGraphics;
var text;
var player;
var showDebug = false;
var groundLayer;
var coinLayer;
var coinsCollected = 0;

function preload ()
{
    this.load.image('ground_1x1',     'assets/tilemaps/tiles/ground_1x1.png');
    this.load.spritesheet('coin',     'assets/sprites/coin.png', { frameWidth: 32, frameHeight: 32 });
    this.load.tilemapTiledJSON('map', 'assets/tilemaps/maps/tile-collision-test.json');
    this.load.image('player',         'assets/sprites/phaser-dude.png');
}

function create ()
{
    map = this.make.tilemap({ key: 'map' });
    var groundTiles = map.addTilesetImage('ground_1x1');
    var coinTiles = map.addTilesetImage('coin');

    // Static layer above dynamic here
    groundLayer = map.createStaticLayer('Ground Layer', groundTiles, 0, 0).setDepth(10);
    coinLayer =   map.createDynamicLayer('Coin Layer',  coinTiles,   0, 0).setDepth(0);

    groundLayer.setCollisionBetween(1, 25);
    coinLayer.setTileIndexCallback(26, hitCoin, this);

    player = this.physics.add.sprite(80, 70, 'player');

    this.physics.add.collider(player, groundLayer);
    this.physics.add.overlap(player, coinLayer);

    this.cameras.main.setBounds(0, 0, map.widthInPixels, map.heightInPixels);
    this.cameras.main.startFollow(player);
    // Small shift to see bug better
    this.cameras.main.x += 200;

    cursors = this.input.keyboard.createCursorKeys();
}

function hitCoin (sprite, tile)
{
    coinLayer.removeTileAt(tile.x, tile.y);

    this.cameras.main.fadeOut(250);

    setTimeout(() => {
        this.cameras.main.fadeIn(250);
    }, 500);

    return false;
}

function update (time, delta)
{
    player.body.setVelocity(0);

    if (cursors.left.isDown)
        player.body.setVelocityX(-200);
    else if (cursors.right.isDown)
        player.body.setVelocityX(200);

    if (cursors.up.isDown)
        player.body.setVelocityY(-200);
    else if (cursors.down.isDown)
        player.body.setVelocityY(200);
}
```

**Comment by c-andrews** [comment]
Im also getting this issue only in WebGL. The cameras overlay looks to be stuck at 0,0 of the level and not attached to the camera. Did you resolve the issue?

**Comment by inmylo** [comment]
@c-andrews , nope. Still waiting for a fix or a workaround advice

**Comment by c-andrews** [comment]
@inmylo, I've not been able to fix it but I've just added a Graphics object and set its depth to the top and it seems to be working. Not the best solution but it works for me.

`
this._overlay = this.add.graphics();
this._overlay.fillStyle(0x000000, 0);
this._overlay.fillRect(0, 0, this.game.scale.width, this.game.scale.height);
this._overlay.setScrollFactor(0);
this._overlay.setDepth(OVERLAY_DEPTH);
`

**Comment by photonstorm** [comment]
I can confirm this was fixed in the 3.23 release. Sorry for not updating this issue with that news.
